heron/schedulers/src/java/com/twitter/heron/scheduler/aurora/AuroraScheduler.java

```diff
@@ -66,14 +67,28 @@ public void initialize(Config mConfig, Config mRuntime) {
    * @return AuroraController
    */
   protected AuroraController getController() {
+    String klass = config.getStringValue(Key.AURORA_CONTROLLER_CLASS);
     Config localConfig = Config.toLocalMode(this.config);
-    return new AuroraCLIController(
-        Runtime.topologyName(runtime),
-        Context.cluster(localConfig),
-        Context.role(localConfig),
-        Context.environ(localConfig),
-        AuroraContext.getHeronAuroraPath(localConfig),
-        Context.verbose(localConfig));
+    switch (klass) {
```

Comment:
This implementation here is really ugly and bad.
One alternative is to use constant enum;
another alternative is to use a boolean to decide which one to use since you have only two cases and it will fail otherwise.
Polymorphism can be another one.

Reply:
changed to boolean